### ISSUE TITLE
How to interpret Stable Baselines logs?

### ISSUE THREAD

**Issue by sophiagu** [hill-a/stable-baselines#1035]
I don't find any documentation about these logs generated by Stable Baselines during training. e.g.:

| approxkl                   | 6.661338e-16   |
| clipfrac                     | 0.0            |
| explained_variance | 0.865          |
| fps                            | 1641           |
| n_updates                | 1362           |
| policy_entropy         | 1.4189396      |
| policy_loss               | -2.9802322e-08 |
| serial_timesteps      | 1362           |
| time_elapsed           | 36.9           |
| total_timesteps       | 54480          |
| value_loss                | 0.00052024494  |

Specifically, how do 'approxkl', 'explained_variance', 'policy_entropy', 'policy_loss' and 'value_loss' tell how good is my current agent doing respectively?

**Comment by araffin** [comment]
Hello,

The question was mostly answered in https://github.com/hill-a/stable-baselines/issues/121#issuecomment-445469720

Short answer: please read more about PPO (cf doc for resources) and look at the code if you want the exact details.
The only value that is not defined is `approxkl` which is the approximate kl divergence between the old and the new policy.

**Comment by sophiagu** [comment]
Thanks for the suggestions! But I notice that many terms of those logs have been changed since the comment you linked.
Also, I've read the stable baselines' doc for PPO but it doesn't directly explain those terms. I feel a few short sentences would be suffice if you could explain the rest of the terms here. Just like how you explained approxkl. That would be helpful. Thanks!

**Comment by araffin** [comment]
>Also, I've read the stable baselines' doc for PPO but it doesn't directly explain those terms. 

As mentioned before, please read the PPO paper or at least the spinning up guide (link in the doc).
I can tell you that `actor_loss` is the loss for the actor ... but it won't help you much, this is related to the algorithm itself.
As written in the README: " Stable Baselines (SB) assumes you have some knowledge about Reinforcement Learning (RL). You should not utilize this library without some practice. To that extent, we provide good resources in the documentation to get started with RL."


>But I notice that many terms of those logs have been changed since the comment you linked.

which ones?

**Comment by sophiagu** [comment]
OK. I found [this doc](http://joschu.net/docs/nuts-and-bolts.pdf) to be helpful. It explains that we want `approxkl` to be small and `policy_entropy` to be large.

That aside, the main reason I asked this question is that I wasn't sure how to deal with very small `policy_loss` but at the same time, both `approxkl` and `policy_entropy` look reasonable.
On one hand, small `policy_loss` seems to mean that my policy is already good (but in my case, the policy is actually bad);
On the other hand, since a policy is only moved by the gradient of `policy_loss`, it therefore seems that I should increase my learning rate in order to compensate for the small gradient (but again, increasing the learning rate doesn't help my policy to learn). 
So is my understanding not correct? Or does having a very small `policy_loss` mean something is wrong with the agent?

**Comment by araffin** [comment]
>Or does having a very small policy_loss mean something is wrong with the agent?

The answer is again in the spinning up guide ;)
https://spinningup.openai.com/en/latest/spinningup/rl_intro3.html

"Even though we describe this as a loss function, it is not a loss function in the typical sense from supervised learning. There are two main differences from standard loss functions. [...]"

**Comment by sophiagu** [comment]
Oh ok. That's an interesting point. ty!
